tox.ini

```
@@ -118,6 +116,8 @@ install_command =
     devdeps: python -I -m pip install -v --pre
 
 commands =
+    mpldev: pip install -U --pre -i https://pypi.anaconda.org/scientific-python-nightly-wheels/simple matplotlib
```

:wave: As noted in the CI logs (https://github.com/astropy/astropy/actions/runs/8078075132/job/22069637834?pr=16122#step:10:33) as there are additional dependencies that don't live on the index you'll need to include the PyPI as the `--extra-index-url`. c.f. the [example in the `scientific-python/upload-nightly-action` README](https://github.com/scientific-python/upload-nightly-action/blob/50053b39cfd4eda8a4046cab060998ea1f8b839e/README.md#using-nightly-builds-in-ci)


```suggestion
    mpldev: pip install -U --pre --index-url https://pypi.anaconda.org/scientific-python-nightly-wheels/simple --extra-index-url https://pypi.org/simple matplotlib
```

---

Oh, that's it ! I got confused while cherry-picking parts to revert. `PIP_EXTRA_INDEX_URL` is already set on line 25, but I was indeed effectively ignoring it here. Thanks !

---

I'd still in general recommend switching the indexes (to `--index-url https://pypi.anaconda.org/scientific-python-nightly-wheels/simple --extra-index-url https://pypi.org/simple`) to get in the habit of helping guard against what happened to [PyTorch in 2022](https://pytorch.org/blog/compromised-nightly-dependency/). But in this particular case it doesn't matter much, given that the name of the packges exist already on both indexes and you are trusting `scientific-python` to not do anything shady on our index.

---

wow, thanks for sharing ! Happy to update this patch if @pllim feels we should

---

Revisiting this comment now... The line in the suggestion no longer exists. In tox, we define `PIP_EXTRA_INDEX_URL` in `setenv`, following by `matplotlib>=0.0.dev0` in `deps`. Not sure if I want to mess with `--index-url` now especially that this is not a concern right now. Thanks anyway!